openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java

```diff
@@ -0,0 +1,91 @@
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.entity.services.ingestionPipelines.IngestionPipeline;
+import org.openmetadata.schema.entity.services.ingestionPipelines.PipelineStatus;
+import org.openmetadata.schema.monitoring.EventMonitorProvider;
+import org.openmetadata.schema.type.ChangeEvent;
+import org.openmetadata.service.util.MicrometerBundleSingleton;
+import software.amazon.awssdk.services.cloudwatch.model.CloudWatchException;
+
+@Slf4j
+public class PrometheusEventMonitor extends EventMonitor {
+
+  private final PrometheusMeterRegistry meterRegistry;
+
+  public PrometheusEventMonitor(
+      EventMonitorProvider eventMonitorProvider, EventMonitorConfiguration config, String clusterPrefix) {
+    super(eventMonitorProvider, config, clusterPrefix);
+    meterRegistry = MicrometerBundleSingleton.prometheusMeterRegistry;
+  }
+
+  @Override
+  protected void pushMetric(ChangeEvent event) {
+    String fqn = event.getEntityFullyQualifiedName();
+    IngestionPipeline ingestionPipeline = (IngestionPipeline) event.getEntity();
+    String pipelineType = ingestionPipeline.getPipelineType().toString();
+
+    try {
+      switch (event.getEventType()) {
+        case ENTITY_DELETED:
+        case ENTITY_SOFT_DELETED:
+        case ENTITY_CREATED:
+          incrementIngestionPipelineCounter(fqn, pipelineType, event.getEventType().value());
+          break;
+        case ENTITY_UPDATED:
+          // we can have multiple updates bundled together
+          event
+              .getChangeDescription()
+              .getFieldsUpdated()
+              .forEach(
+                  change -> {
+                    if (change.getName().equals("pipelineStatus") && change.getNewValue() != null) {
+                      PipelineStatus pipelineStatus = (PipelineStatus) change.getNewValue();
+                      incrementIngestionPipelineCounter(fqn, pipelineType, pipelineStatus.getPipelineState().value());
+                    }
+                  });
+
+        default:
+          throw new IllegalArgumentException("Invalid EventType " + event.getEventType());
+      }
+    } catch (IllegalArgumentException | CloudWatchException e) {
+      LOG.error("Failed to publish IngestionPipeline Cloudwatch metric due to " + e.getMessage());
+    }
+  }
+
+  @Override
+  protected void close() {
+    meterRegistry.close();
+  }
+
+  /**
+   * A new counter will be created only if it does not exist for the
+   * given set of tags. Otherwise, micrometer will increase the count
+   * of the existing counter.
+   * Ref <a href="https://stackoverflow.com/questions/59592118/dynamic-tag-values-for-the-counter-metric-in-micrometer">...</a>
+   *
+   * @param fqn Pipeline FQN
+   * @param pipelineType Metadata, Profiler,...
+   * @param eventType running, failed, entityCreated,...
+   */
+  public void incrementIngestionPipelineCounter(String fqn, String pipelineType, String eventType) {
```

Comment:
dynamically building the counter based on the Change Event

Reply:
@pmbrull looks great. Do we have any other data capturing here, like no.of entities that are successfully ingested vs failed etc..

Reply:
also any strings such as "pipelineStatus" lets make private final static

Reply:
We're not yet sending the `summary` of the workflow as a pipeline status. Once we apply that change, we can update the Change Events to track those statistics as well. Let me open a separate ticket for that